src/mutation_data.cpp

```diff
@@ -567,13 +567,51 @@ std::string mutation_branch::desc() const
     return raw_desc.translated();
 }
 
+static bool has_cyclic_dependency( const trait_id &mid, std::vector<trait_id> already_visited )
+{
+    if( contains_trait( already_visited, mid ) ) {
+        return true; // Circular dependency was found.
+    }
+
+    already_visited.push_back( mid );
+
+    // Perform Depth-first search
+    for( const trait_id &current_mutation : mid->prereqs ) {
+        if( has_cyclic_dependency( current_mutation->id, already_visited ) ) {
+            return true;
+        }
+    }
+
+    for( const trait_id &current_mutation : mid->prereqs2 ) {
+        if( has_cyclic_dependency( current_mutation->id, already_visited ) ) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+static void check_has_cyclic_dependency( const trait_id &mid )
+{
+    std::vector<trait_id> already_visited;
+    if( ::has_cyclic_dependency( mid, already_visited ) ) {
+        debugmsg( "mutation %s references itself in either of it's prerequsition branches.  Program will crash if player would gain this mutation.",
+                  mid.c_str() );
+    }
+}
+
 static void check_consistency( const std::vector<trait_id> &mvec, const trait_id &mid,
                                const std::string &what )
 {
     for( const auto &m : mvec ) {
         if( !m.is_valid() ) {
             debugmsg( "mutation %s refers to undefined %s %s", mid.c_str(), what.c_str(), m.c_str() );
         }
+
+        if( m == mid ) {
+            debugmsg( "mutation %s refers to itself in %s context.  Program will crash if player would gain this mutation",
```

Comment:
Minor rephrasing: "Mutation %s references itself in %s context. The program will crash if the player gains this mutation."